routers/api/v1/user/gpg_key.go

```diff
@@ -121,6 +121,10 @@ func GetGPGKey(ctx *context.APIContext) {
 		}
 		return
 	}
+	if !ctx.Doer.IsAdmin && key.OwnerID != ctx.Doer.ID {
```

Comment:
Why it checkes `IsAdmin`? I do not see how it makes sense.

Could it just use something like `GetGPGKeyForUserByID()`?